torch/csrc/api/include/torch/tensor.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <torch/csrc/autograd/variable.h>
+
+namespace at {
+struct Type;
+enum class Backend;
+enum class ScalarType;
+} // namespace at
+
+namespace torch {
+/// There will only be gradient recording tensors in the frontend API.
+using Tensor = autograd::Variable;
```

Comment:
I'm still a bit worried that people will start mixing up tensor types (`torch::Tensor` and `at::Tensor), especially that they are implicitly convertible

Reply:
This was just to present my idea that there will only be `torch::Tensor` in this API. Users should neither know about the concept of `at::Tensor` nor `autograd::Variable`. We will have to be careful when migrating existing users like StarCraft, but for new users there should only be `torch::Tensor` and nothing else (like in Python)

Reply:
That's impossible, unless `torch::` will become an exact mirror of `at::`. Even you used `at::tanh` in your `LSTM` implementation, but this means that you silently introduced an `at::Tensor` into the mix.

Reply:
I was thinking of doing `using namespace torch { using namespace at; using Tensor = autograd::Variable; }`.

Reply:
I see, that makes sense

Reply:
Can it be reconciled with the API that jit right now takes (seemingly enforcing everything to be variables too): https://sourcegraph.com/github.com/pytorch/pytorch/-/blob/torch/csrc/jit/variable_tensor_list.h#L8:8